### PR TITLE
Clarify and correct the description of EdgeQL query evaluation semantics

### PR DESCRIPTION
**Pull request by elprans**

### CODE REVIEW

File: doc/edgeql/expressions/paths.rst [hunk 1]
```diff
@@ -6,9 +6,9 @@
 Paths
 =====
 
-A *path expression* (or simply a *path*) is a special kind of set reference.
-It represents a set of values that are reachable when traversing a given
-sequence of links or properties from some source set.
+A *path expression* (or simply a *path*) represents a set of values that are
+reachable when traversing a given sequence of links or properties from some
```
Comment (1st1) [inline doc/edgeql/expressions/paths.rst]: A couple of paragraphs below this line: can we change "concatenation" to "collection" or something else in "[..] then it represents a concatenation of property values [..]"? Concatenation usually implies strings and thus it takes another additional read to disambiguate.

File: doc/edgeql/overview.rst [hunk 2]
```diff
@@ -157,66 +149,64 @@ An implicit ``SELECT`` subquery is assumed in the following situations:
...
-   - make another cartesian product from the input tuple and the
-     sets produced by subqueries and functions, excluding the sets
-     used as aggregate arguments;
+      SELECT (
+        User.friends.firstname,
+        User.friends.lastname,
+        Issue.priority.name,
+        Issue.number,
+        Status.name
+      )
```
Comment (1st1) [inline doc/edgeql/overview.rst]: Add ";"

File: doc/edgeql/overview.rst [hunk 3]
```diff
@@ -157,66 +149,64 @@ An implicit ``SELECT`` subquery is assumed in the following situations:
 - the right side of turnstile (``:=``) in expression aliases and
   :ref:`shape element declarations <ref_eql_expr_shapes>`;
 
-- the majority of statement clauses;
+- the majority of statement clauses.
 
-- any set returning function or operator (e.g. a :ref:`set constructor
-  <ref_eql_expr_index_set_ctor>`).
+A nested query is called a *subquery*.  The phrase "*in the query*"
```
Comment (1st1) [inline doc/edgeql/overview.rst]: .. In this section, the phrase "in the query" ..

File: doc/edgeql/overview.rst [hunk 4]
```diff
@@ -157,66 +149,64 @@ An implicit ``SELECT`` subquery is assumed in the following situations:
...
-     the value of the query and store the result.
+   In the above query, the longest common prefixes are: ``User.friends``,
+   ``Issue``, and ``Status.name``.
 
-4. Append the results of all iterations to obtain the final result set.
+3. Create a set of *input tuples* as a cartesian product of the input sets
+   using the following rules:
 
+   - if the query input list is empty, the input tuple set would
+     contain a single empty tuple;
```
Comment (1st1) [inline doc/edgeql/overview.rst]: "a single empty tuple" -> "a single empty input tuple"

File: doc/edgeql/overview.rst [hunk 5]
```diff
@@ -157,66 +149,64 @@ An implicit ``SELECT`` subquery is assumed in the following situations:
...
-2. If *any* of the functions in the query have arguments declared as
-   ``OPTIONAL``, these functions are called as usual, with arguments
-   passed as empty sets.
+     * subqueries are evaluated recursively from step 1;
 
-For example, the following query returns an empty set:
+     * a function or an operator is evaluated in a loop over a cartesian
+       product of its non-aggregate arguments
+       (empty ``OPTIONAL`` arguments are excluded from the product),
+       aggregate arguments are passed as a whole set; the results of the
```
Comment (1st1) [inline doc/edgeql/overview.rst]: I'd have a separate bullet for "aggregate arguments are passed as a whole set"

File: doc/edgeql/expressions/paths.rst (hunk 1 above)
Comment (elprans) [reply doc/edgeql/expressions/paths.rst]: Fixed.

**Comment by vpetrovykh** [comment]
"For example a reference to the User object type in the following query will result in a set of all User objects"
Maybe it should be *will resolve to* instead of *will result in*.

**Comment by vpetrovykh** [comment]
"An aggregate parameter is normally specified using the SET OF modifier"
Why "normally"? If we don't have any other way, "normally" should be dropped.

**Pull request merged**